===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ p11_asn1_cache_take:n$4 == { 0 } }) free(*(p11_asn1_cache_take:item)) at 5 (line 356, column 7) }
===== Actual patch in diff format =====
@@ -354,6 +354,7 @@ p11_asn1_cache_take (p11_asn1_cache *cache,
 	}
 
 	if (!p11_dict_set (cache->items, (void *)der, item))
+		free(item);
 		return_if_reached ();
 }
 
